File: src/main/java/com/leomadrassi/trollingfreedomreborn/main/DynamicAlias.java
package com.leomadrassi.trollingfreedomreborn.main;

import org.bukkit.command.Command;
import org.bukkit.command.CommandSender;
import org.bukkit.command.PluginCommand;
import java.util.List;

public class DynamicAlias extends Command {
    private final PluginCommand original;

    protected DynamicAlias(String name, PluginCommand original) {
        super(name);
        this.original = original;
        // Copy description and usage from original
        this.setDescription(original.getDescription());
        this.setUsage(original.getUsage());
    }

    @Override
    public boolean execute(CommandSender sender, String commandLabel, String[] args) {
        // Redirect execution to the original command's executor
        return original.execute(sender, commandLabel, args);
    }

    @Override
    public List<String> tabComplete(CommandSender sender, String alias, String[] args) throws IllegalArgumentException {
        // Redirect tab completion to the original command's tab completer
        return original.tabComplete(sender, alias, args);
    }
}

File: src/main/java/com/leomadrassi/trollingfreedomreborn/main/DynamicAlias.kt
package com.leomadrassi.trollingfreedomreborn.main

import org.bukkit.command.Command
import org.bukkit.command.CommandSender
import org.bukkit.command.PluginCommand

class DynamicAlias(name: String, private val original: PluginCommand) : Command(name) {

    init {
        description = original.description
        usage = original.usage
    }

    override fun execute(sender: CommandSender, commandLabel: String, args: Array<String>): Boolean {
        return original.execute(sender, commandLabel, args)
    }

    override fun tabComplete(sender: CommandSender, alias: String, args: Array<String>): MutableList<String> {
        return original.tabComplete(sender, alias, args) as MutableList<String>? ?: mutableListOf()
    }
}
